src/edit.c

```diff
@@ -2849,7 +2855,6 @@ cursor_down_inner(win_T *wp, long n)
 	// count each sequence of folded lines as one logical line
 	while (n--)
 	{
-	    // Move to last line of fold, will fail if it's the end-of-file.
```

Comment:
This regressed in #12480. With this context, I think this comment was moved erroneously and is out of place here. New comment above explains what is done such that the outer `cursor_down()` fails again in that case.

Reply:
Hmm, I think this needs to be handled before entering the inner function. So I moved it there and added additional test cases.